application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -280,6 +326,66 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContent, @Nullable String previousContent) {
+
+        if (List
+            .of(Subcommand.CREATE, Subcommand.CREATE_WITH_MESSAGE, Subcommand.EDIT,
+                    Subcommand.EDIT_WITH_MESSAGE)
+            .contains(subcommand) && newContent == null) {
+            logger.debug("newContent is null even though the subcommand should supply a value.");
+            return;
+        }
+
+        if (List.of(Subcommand.EDIT, Subcommand.EDIT_WITH_MESSAGE, Subcommand.DELETE)
+            .contains(subcommand) && previousContent == null) {
+            logger
+                .debug("previousContent is null even though the subcommand should supply a value.");
+            return;
+        }
+
+        // to suppress warning "Argument '' might be null"
+        if (newContent == null) {
+            newContent = "";
+        }
+        if (previousContent == null) {
+            previousContent = "";
+        }
```

Comment:
Thats the wrong way of handling this. If your previous `if`s make it impossible that they are `null` later down in the code, you should instead use `Objects.requireNonNull(newContent)` in the code down below.

That way you get fail-fast. I.e. if for some reason your code logic is buggy and a null still sneaked in, it will crash and we notice it quick.

With your current code we would not notice at all.

Reply:
done.